tutorials/plot_brainstorm_auditory.py

```diff
@@ -0,0 +1,350 @@
+# 170 ms in both conditions but much stronger in the standard condition. P300
+# is visible in deviant condition only (decision making in preparation of the
+# button press). You can view the topographies from a certain time span by
+# painting an area with clicking and holding the left mouse button.
+evoked_std.plot(window_title='Standard', gfp=True)
+evoked_dev.plot(window_title='Deviant', gfp=True)
+
+###############################################################################
+# Show activations as topography figures.
+times = np.arange(0.05, 0.301, 0.025)
+evoked_std.plot_topomap(times=times, title='Standard')
+evoked_dev.plot_topomap(times=times, title='Deviant')
+
+###############################################################################
+# We can see the MMN effect more clearly by looking at the difference between
+# the two conditions. P50 and N100 are no longer visible, but MMN/P200 and
+# P300 are emphasised.
+evoked_difference = evoked_std.copy()
+evoked_difference.data = evoked_dev.data - evoked_std.data
+evoked_difference.plot(window_title='Difference', gfp=True)
+
+###############################################################################
+# Source estimation.
+# We compute the noise covariance matrix from the empty room measurement
+# and use it for the other runs.
+reject = dict(mag=4e-12)
+cov = mne.compute_raw_covariance(raw_erm, reject=reject)
+cov.plot(raw_erm.info)
+del raw_erm
+
+###############################################################################
+# The transformation is read from a file. More information about coregistering
+# the data, see :ref:`ch_interactive_analysis` or
+# :func:`mne.gui.coregistration`.
+trans_fname = op.join(data_path, 'MEG', 'bst_auditory',
+                      'bst_auditory-trans.fif')
+trans = mne.read_trans(trans_fname)
+
+###############################################################################
+# To save time and memory, the forward solution is read from a file. Set
+# ``use_precomputed=False`` in the beginning of this script to build the
+# forward solution from scratch. The head surfaces for constructing a BEM
+# solution are read from a file. Since the data only contains MEG channels, we
+# only need the inner skull surface for making the forward solution. For more
+# information: :ref:`CHDBBCEJ`, :class:`mne.setup_source_space`,
+# :ref:`create_bem_model`, :func:`mne.bem.make_watershed_bem`.
+if use_precomputed:
+    fwd_fname = op.join(data_path, 'MEG', 'bst_auditory',
+                        'bst_auditory-meg-oct-6-fwd.fif')
+    fwd = mne.read_forward_solution(fwd_fname)
+else:
+    src = mne.setup_source_space(subject, subjects_dir=subjects_dir,
+                                 overwrite=True)
+    surfs_fname = op.join(subjects_dir, 'bst_auditory', 'bem',
+                          'bst_auditory-inner_skull.fif')
+    surfs = mne.read_bem_surfaces(surfs_fname)
+
+    bem = mne.make_bem_solution(surfs)
```

Comment:
The current code assumes a standard FreeSurfer recon. If we want to allow for non-FreeSurfer data, then sure an arg list is fine. The problem arises that .surf files don't say what surface they actually *are* (they just contain a standard mesh definition), so users could do all sorts of things wrong (e.g., make a single-layer BEM from the head surface instead of inner skull surface) accidentally by passing in a list of surfs. `.fif` files shouldn't suffer from this problem since they contain the surface ID. Not sure what the cleanest solution is.